onboard/nucleo_bridge/Frontend.h

```diff
@@ -0,0 +1,197 @@
+	    (*controllers)["RA_4"]->config(msg->Kp[4], msg->Ki[4], msg->Kd[4]);
+	    (*controllers)["RA_5"]->config(msg->Kp[5], msg->Ki[5], msg->Kd[5]);*/
+	}
+	    
+	void sa_config_cmd(LCM_INPUT, const SAConfigCmd *msg){
+	    (*controllers)["SA_0"]->config(msg->Kp[0], msg->Ki[0], msg->Kd[0]);
+	    (*controllers)["SA_1"]->config(msg->Kp[1], msg->Ki[1], msg->Kd[1]);
+	    (*controllers)["SA_2"]->config(msg->Kp[2], msg->Ki[2], msg->Kd[2]);
+	}
+	    
+	void hand_openloop_cmd(LCM_INPUT, const HandCmd *msg){
+	    (*controllers)["HAND_FINGER_POS"]->open_loop(msg->finger);
+	    (*controllers)["HAND_FINGER_NEG"]->open_loop(msg->finger);
+	    (*controllers)["HAND_GRIP_POS"]->open_loop(msg->grip);
+	    (*controllers)["HAND_GRIP_NEG"]->open_loop(msg->grip);
+	}
+
+	void gimbal_cmd(LCM_INPUT, const GimbalCmd *msg){
+	    (*controllers)["GIMBAL_PITCH_0_POS"]->open_loop(msg->pitch[0]);
+	    (*controllers)["GIMBAL_PITCH_0_NEG"]->open_loop(msg->pitch[0]);
+	    (*controllers)["GIMBAL_PITCH_1_POS"]->open_loop(msg->pitch[1]);
+	    (*controllers)["GIMBAL_PITCH_1_NEG"]->open_loop(msg->pitch[1]);
+	    (*controllers)["GIMBAL_YAW_0_POS"]->open_loop(msg->yaw[0]);
+	    (*controllers)["GIMBAL_YAW_0_NEG"]->open_loop(msg->yaw[0]);
+	    (*controllers)["GIMBAL_YAW_1_POS"]->open_loop(msg->yaw[1]);
+	    (*controllers)["GIMBAL_YAW_1_NEG"]->open_loop(msg->yaw[1]);
+	}
+	
+	void refreshAngles() {
+	    (*controllers)["RA_0"]->angle();
+	    (*controllers)["RA_1"]->angle();
+	    (*controllers)["RA_2"]->angle();
+	    (*controllers)["RA_3"]->angle();
+	    (*controllers)["RA_4"]->angle();
+	    (*controllers)["RA_5"]->angle();
+	    (*controllers)["SA_0"]->angle();
+	    (*controllers)["SA_1"]->angle();
+	    (*controllers)["SA_2"]->angle();
+	}
+
+	void ra_pos_data(){
+		ArmPosition msg;
+	    msg.joint_a = (*controllers)["RA_0"]->currentAngle;
+	    msg.joint_b = (*controllers)["RA_1"]->currentAngle;
+	    msg.joint_c = (*controllers)["RA_2"]->currentAngle;
+	    msg.joint_d = (*controllers)["RA_3"]->currentAngle;
+	    msg.joint_e = (*controllers)["RA_4"]->currentAngle;
+	    msg.joint_f = (*controllers)["RA_5"]->currentAngle;
+	    lcmBus->publish("/arm_position", &msg);
+
+
+	    lastTime = NOW;
+	}
+
+	void sa_pos_data(){
+	    SAPosData msg;
+	    msg.angle[0] = (*controllers)["SA_0"]->currentAngle;
+	    msg.angle[1] = (*controllers)["SA_1"]->currentAngle;
+	    msg.angle[2] = (*controllers)["SA_2"]->currentAngle;
+	    lcmBus->publish("/sa_pos_data", &msg);
+
+	    lastTime = NOW;
+	}
+
+	void sa_zero_trigger(LCM_INPUT, const SAZeroTrigger *msg){
+	    (*controllers)["SA_0"]->zero();
+	    (*controllers)["SA_1"]->zero();
+	    (*controllers)["SA_2"]->zero();
+	}
+
+/*	void ra_zero_trigger(LCM_INPUT, const RAZeroTrigger *msg){
+		(*controllers)["RA_0"]->zero();
+		(*controllers)["RA_1"]->zero();
+		(*controllers)["RA_2"]->zero();
+		(*controllers)["RA_3"]->zero();
+		(*controllers)["RA_4"]->zero();
+		(*controllers)["RA_5"]->zero();
+	}*/
```

Comment:
Remove?

Reply:
This could be important when we activate IK on the 2021 arm